File: kurs programowania/lista3/cpp/Kolo.cpp
#include "Kolo.hpp"

#include <iostream>

#define PI 3.14159265358979323846

Kolo::Kolo(int r)
{
    if (r < 0)
        throw std::runtime_error("illegal_args");

    this->r = r;
}

double Kolo::pole()
{
    return PI * r * r;
}

double Kolo::obwod()
{
    return 2 * PI * r;
}

File: kurs programowania/lista3/zadanie1/Kolo.java
public class Kolo extends Figura {

    public int promien;

    public Kolo(int promien) {

        if (promien < 0)
            throw new IllegalArgumentException();

        this.promien = promien;
    }

    public double pole() {

        return Math.PI * promien * promien;
    }

    public double obwod() {

        return 2 * Math.PI * promien;
    }
}